### PR TITLE
Adds FleetDM integration guide

### PR DESCRIPTION
**Pull request by ZPain8464**
This PR adds a FleetDM integration guide at `/docs/integrations/fleet`. 

Resolves https://github.com/pomerium/documentation/issues/1627

### CODE REVIEW

File: content/docs/integrations/fleet.mdx [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+- Deploy a Fleet instance.
+- Configure the FleetDM plugin in Pomerium Enterprise.
+- Create authorization policies in Pomerium with FleetDM criteria.
+
+To complete this guide, you need:
+
+- [Pomerium Enterprise](https://www.pomerium.com/docs/enterprise)
+- [Pomerium Core](https://www.pomerium.com/docs/core)
+- A deployed [Fleet instance](https://github.com/fleetdm/fleet/tree/main/infrastructure/render)
+- A configured certificate authority (CA) and client certificate
```
[the remark this answers is not in the document]
Comment (wasaga) [reply content/docs/integrations/fleet.mdx:151]: mkcert is something generic people can reproduce themselves but we need make clear this is not for production usage

File: content/docs/integrations/fleet.mdx [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+
+- Deploy a Fleet instance.
+- Configure the FleetDM plugin in Pomerium Enterprise.
+- Create authorization policies in Pomerium with FleetDM criteria.
+
+To complete this guide, you need:
+
+- [Pomerium Enterprise](https://www.pomerium.com/docs/enterprise)
+- [Pomerium Core](https://www.pomerium.com/docs/core)
+- A deployed [Fleet instance](https://github.com/fleetdm/fleet/tree/main/infrastructure/render)
```
[the remark this answers is not in the document]
Comment (wasaga) [reply content/docs/integrations/fleet.mdx:150]: Could be but it's out of scope how FleetDM is set up

File: content/docs/integrations/fleet.mdx [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+- A configured certificate authority (CA) and client certificate
+- An environment to install a Fleet agent
+
+:::note
+
+For the purposes of this guide, we recommend installing the Fleet agent in a VM running macOS. You can install and run a macOS VM for free with Tart. See the [Tart Quickstart](https://tart.run/quick-start/) for more information.
+
+:::
+
+### Configure FleetDM
```
Comment (wasaga) [inline content/docs/integrations/fleet.mdx:160]: Let's keep FleetDM config out of scope for this guide

File: content/docs/integrations/fleet.mdx [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+   To add other devices to Fleet, distribute fleetd using Chef, Ansible, Jamf, or Puppet. Learn how: https://fleetdm.com/learn-more-about/enrolling-hosts
+   ```
+
+1. Run your macOS VM and install the `fleet-osquery.pkg`.
+
+   If you check your Fleet dashboard, you should see the VM listed as an enrolled host:
+
+   ![Reviewing the enrolled host in the Fleet dashboard](./img/fleet/fleet-enrolled-host-vm.png)
+
+1. Lastly, install the CA bundle, which includes the client certificate, on the VM where you installed the Fleet agent.
```
Comment (wasaga) [inline content/docs/integrations/fleet.mdx:203]: Please add the command (security import ...)

Also we're not installing ÇA bundle but just the client certificate.

File: content/docs/integrations/fleet.mdx [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+
+### Configure certificate query
+
+Fleet agents provide a [certificates](https://fleetdm.com/tables/certificates) query that scans a device’s local key store for installed CA bundles. This query exposes certificates data, including the client certificate SHA-1 fingerprint installed on an enrolled host.
```
Comment (wasaga) [inline content/docs/integrations/fleet.mdx:207]:
```suggestion
Fleet agents provide a [certificates](https://fleetdm.com/tables/certificates) query that scans a device’s local key store for installed certificates. This query exposes certificates data, including the client certificate SHA-1 fingerprint installed on an enrolled host.
```

**Review by wasaga — COMMENTED** [review]
We also need add a note somewhere that this is not a production configuration as it uses mkcert. The production configuration requires Private CA integration with FleetDM which is out of scope for this guide.

File: content/docs/integrations/fleet.mdx [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+FleetDM provides an open-source, scalable device management (MDM) solution that supports all major operating systems, including macOS, Linux, and Windows.
+
+The FleetDM integration with Pomerium Enterprise enables you to build policies in Pomerium that express Fleet conditions as Pomerium policy language (PPL).
+
+When Pomerium receives a request from a Fleet-managed device, Pomerium will make an authorization decision based on the device’s state and the applied policy criteria.
+
+## How Pomerium works with FleetDM
+
+Fleet provides a dedicated web service (Fleet UI) where administrators can manage teams, issued devices (”enrolled hosts”), Fleet policies, and more.
```
Comment (wasaga) [inline content/docs/integrations/fleet.mdx:16]: the product name is `FleetDM` - please also check in few places below

File: content/docs/integrations/fleet.mdx [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+
+In Fleet UI, administrators can install Fleet agents on remote hosts. The Fleet agents periodically query hosts to gather updated host data, and push the information to Fleet servers. Administrators can review reports in Fleet UI to monitor hosts for security vulnerabilities.
+
+Pomerium can integrate your Fleet instance as an external data source (the “FleetDM plugin”) in the Enterprise Console. You can use custom PPL criteria to build policies in Pomerium that enforce Fleet conditions before authorizing or denying access to services behind Pomerium. For example, Pomerium can deny access if:
+
+- A device fails a [policy](https://fleetdm.com/docs/rest-api/rest-api#policies) defined in Fleet.
```
Comment (wasaga) [inline content/docs/integrations/fleet.mdx:22]:
```suggestion
- A device fails a [policy](https://fleetdm.com/securing/what-are-fleet-policies) defined in Fleet.
```

File: content/docs/integrations/fleet.mdx [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+- A device is affected by a [vulnerability](https://fleetdm.com/docs/rest-api/rest-api#vulnerabilities).
```
Comment (wasaga) [inline content/docs/integrations/fleet.mdx:23]:
```suggestion
- A device is affected by a [vulnerability](https://fleetdm.com/guides/vulnerability-processing).
```

File: content/docs/integrations/fleet.mdx [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+
+- `fleetdm.com/CertificateSHA1Fingerprint`: represents a client certificate SHA-1 fingerprint that maps to a Fleet host ID
+- `fleetdm.com/Host`: represents the state of a Fleet host
+- `fleetdm.com/Policy`: represents a Fleet policy
+
+Pomerium matches an incoming request with a Fleet-managed host by requiring the user to present a valid client certificate in the browser.
+
+Pomerium receives the client certificate, generates its SHA-1 fingerprint, and matches it with the host ID stored in the FleetDM plugin.
+
+Using the host ID, Pomerium can periodically fetch [host](https://fleetdm.com/docs/rest-api/rest-api#example33) data from Fleet. The host data provides an updated [report](https://fleetdm.com/docs/rest-api/rest-api#get-query-report) of a host’s state, including its compliance to Fleet policies and vulnerability status.
```
Comment (wasaga) [inline content/docs/integrations/fleet.mdx:37]:
```suggestion
Pomerium Enterprise continuously syncs [host](https://fleetdm.com/docs/rest-api/rest-api#example33) data from FleetDM. The host data represents its compliance to Fleet policies and exposure to vulnerabilities.
```

File: content/docs/integrations/fleet.mdx [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+Pomerium provides a custom FleetDM criterion that points to records collected from the FleetDM plugin.
+
+Custom FleetDM operators enable you to build policies that enforce Fleet conditions. Only [deny](/docs/capabilities/ppl#actions) actions are supported.
+
+Custom FleetDM operators include:
+
+| **Key name** | **Value type** | **Description** |
+| --- | --- | --- |
+| `last_seen_over` | string | Denies access if a Fleet agent hasn’t updated host data in the specified time. |
+| `vulnerabilities` | string | Denies access if a host is affected by a specific CVE. |
```
Comment (wasaga) [inline content/docs/integrations/fleet.mdx:50]:
```suggestion
| `vulnerabilities` | array(string) | Denies access if a host is affected by a specific CVE. |
```

File: content/docs/integrations/fleet.mdx [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+| `failing_policies` | string | Denies access if a host fails a specific Fleet policy. |
```
Comment (wasaga) [inline content/docs/integrations/fleet.mdx:51]:
```suggestion
| `failing_policies` | array(int) | Denies access if a host fails a specific Fleet policy. |
```

File: content/docs/integrations/fleet.mdx [hunk 12]
```diff
@@ -0,0 +1,195 @@
...
+
+![Creating a Fleet policy in the Fleet dashboard](./img/fleet/fleet-policy.png)
+
+:::note
+
+See the Fleet [policies](https://fleetdm.com/docs/rest-api/rest-api#policies) documentation to learn about how policies work.
+
+:::
+
+### Configure Pomerium Enterprise
```
Comment (wasaga) [inline content/docs/integrations/fleet.mdx:167]: before that, they need run `pomerium/datasource:main` with args 

```
        - fleetdm
        - --api-url=https://pomerium.cloud.fleetdm.com/
        - --api-token=THEIR_FLEETDM_API_TOKEN
        - --cert-query-id=THEIR_FLEETDM_CERT_QUERY_ID
        - --address=:8080
```

**Comment by ZPain8464** [comment]
@wasaga I left a note at the beginning of the guide portion of the page. It makes it pretty clear this is a testing configuration not intended for production use. Let me know if it needs any changes.

File: content/docs/integrations/fleetdm.mdx [hunk 13]
```diff
@@ -0,0 +1,230 @@
...
+   ```
+
+In the Enterprise Console:
+
+1. Select **External Data**.
+1. Select **ADD EXTERNAL DATA SOURCE**.
+1. In the **URL** field, enter the URL to the external data source.
+1. Select **SAVE EXTERNAL DATA SOURCE**.
+
+### Create a policy in Pomerium
```
Comment (wasaga) [inline content/docs/integrations/fleetdm.mdx:210]: we don't seem to explain that the 

- user has to import the CA `mkcert --CAROOT`/rootCA.pem 
- user has to create a new route they want to protect
- user has to select Downstream CA and the uploaded certificate

**Review by wasaga — APPROVED** [review]

**Pull request merged**